projects/ADuCM3029_demo_cn0357/src/cn0357_app.cpp

```diff
@@ -0,0 +1,395 @@
+  - Redistributions of source code must retain the above copyright notice,
+    this list of conditions and the following disclaimer.
+  - Redistributions in binary form must reproduce the above copyright notice,
+    this list of conditions and the following disclaimer in the documentation
+    and/or other materials provided with the distribution.
+  - Modified versions of the software must be conspicuously marked as such.
+  - This software is licensed solely and exclusively for use with processors
+    manufactured by or for Analog Devices, Inc.
+  - This software may not be combined or merged with other code in any manner
+    that would cause the software to become subject to terms and conditions
+    which differ from those listed here.
+  - Neither the name of Analog Devices, Inc. nor the names of its
+    contributors may be used to endorse or promote products derived
+    from this software without specific prior written permission.
+  - The use of this software may or may not infringe the patent rights of one
+    or more patent holders.  This license does not release you from the
+    requirement that you obtain separate licenses from these patent holders
+    to use this software.
+
+THIS SOFTWARE IS PROVIDED BY ANALOG DEVICES, INC. AND CONTRIBUTORS "AS IS"
+AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, NON-
+INFRINGEMENT, TITLE, MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+DISCLAIMED. IN NO EVENT SHALL ANALOG DEVICES, INC. OR CONTRIBUTORS BE LIABLE
+FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, PUNITIVE OR
+CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, DAMAGES ARISING OUT OF
+CLAIMS OF INTELLECTUAL PROPERTY RIGHTS INFRINGEMENT; PROCUREMENT OF
+SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+POSSIBILITY OF SUCH DAMAGE.
+
+*****************************************************************************/
+#include "cn0357_app.h"
+
+#include <common.h>
+#include <gas/adi_gas.h>
+#include <drivers/pwr/adi_pwr.h>
+#include <radio/adi_ble_radio.h>
+#include <common/adi_timestamp.h>
+#include <gas/cn0357/adi_cn0357.h>
+#include <common/adi_error_handling.h>
+#include <framework/noos/adi_ble_noos.h>
+#include <base_sensor/adi_sensor_packet.h>
+#include <base_sensor/adi_sensor_errors.h>
+
+using namespace adi_sensor_swpack;
+
+/* Peripheral Advertisement Mode */
+#define PERIPHERAL_ADV_MODE ((ADI_BLE_GAP_MODE)(ADI_BLE_GAP_MODE_CONNECTABLE | ADI_BLE_GAP_MODE_DISCOVERABLE))
+
+#if ADI_APP_USE_BLUETOOTH == 1u
+/* Global Data */
+static bool               gbConnected;
+static ADI_BLE_GAP_MODE   geMode;
+static ADI_BLER_CONN_INFO sConnInfo;
+uint8_t * reg_flag = (uint8_t *)malloc(sizeof(uint8_t));
```

Comment:
There is no need to cast the pointer returned by malloc manually because void pointers are safely promoted to any type of pointer and reduces the risk of casting to a wrong type, which will not always throw a warning.